src/comparearrays.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Compares how given arrays relate to each other. One array can be: same as another array, prefix of another array
+ * or completely different. If arrays are different, first index at which they differ is returned. Otherwise,
+ * a flag specifying the relation is returned. Flags are negative numbers, so whenever a number >= 0 is returned
+ * it means that arrays differ.
+ *
+ *		compareArrays( [ 0, 2 ], [ 0, 2 ] );		// 'SAME'
+ *		compareArrays( [ 0, 2 ], [ 0, 2, 1 ] );		// 'PREFIX'
+ *		compareArrays( [ 0, 2 ], [ 0 ] );			// 'EXTENSION'
+ *		compareArrays( [ 0, 2 ], [ 1, 2 ] );		// 0
+ *		compareArrays( [ 0, 2 ], [ 0, 1 ] );		// 1
+ *
+ * @memberOf utils
+ * @param {Array} a Array that is compared.
+ * @param {Array} b Array to compare with.
+ * @returns {utils.ArrayRelation} How array `a` is related to `b`.
+ */
+export default function( a, b ) {
```

Comment:
Let's better make them named functions. So `export default function fooBar() { ... }`. Like here: https://github.com/ckeditor/ckeditor5-utils/blob/master/src/dom/indexof.js#L15. We'll be consistent and it will be easier to debug.